infrastructure-playbooks/rolling_update.yml

```diff
@@ -83,40 +84,71 @@
   become: True
   tasks:
     - name: set mon_host_count
-      set_fact: mon_host_count={{ groups[mon_group_name] | length }}
+      set_fact:
+        mon_host_count: "{{ groups[mon_group_name] | length }}"
 
     - name: fail when less than three monitors
       fail:
         msg: "Upgrade of cluster with less than three monitors is not supported."
       when:
         - mon_host_count | int < 3
 
+    - name: select a running monitor
+      set_fact:
+        mon_host: "{{ groups[mon_group_name] | difference([inventory_hostname]) | last }}"
+
+    - name: set osd flags
+      command: ceph --cluster {{ cluster }} osd set {{ item }}
+      with_items:
+        - noout
+        - norebalance
+      when:
+        - inventory_hostname == groups[mon_group_name][0]
+        - not containerized_deployment
+
+    - name: set containerized osd flags
+      command: >
+        {{ container_binary }} exec ceph-mon-{{ ansible_hostname }} ceph --cluster {{ cluster }} osd set {{ item }}
```

Comment:
container_binary isn't defined because we don't run ceph-facts role at this point